modules/search/class.jetpack-search.php

```diff
@@ -476,7 +476,7 @@ public function filter__posts_pre_query( $posts, $query ) {
 		$posts_query = new WP_Query( $args );
 
 		// WP Core doesn't call the set_found_posts and its filters when filtering posts_pre_query like we do, so need to do these manually.
-		$query->found_posts   = $this->found_posts;
+		$query->found_posts   = $posts_query->post_count;
```

Comment:
Ok this won't work, my mistake this will always be the per page count. However, the underlying issue does still exist (count being off if search index is out of sync). So any opinion on how to solve would be good!

Reply:
Oh yeah good point didn't consider this; yeah I guess there's really no way to solve this, after all if out index is not up to date then there's no way for us to know how many posts should be returned by the search. We can check if the results returned is in the MySQL DB but unless we return all matching posts for every query there's no way to fix this issue.